src/main/java/lotto/domain/generator/LottoAutoNumberGenerator.java

```diff
@@ -0,0 +1,28 @@
+package lotto.domain.generator;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+import lotto.domain.Lotto;
+import lotto.domain.Number;
+
+public class LottoAutoNumberGenerator implements LottoNumberGenerator {
+
+    private final List<Number> candidateNumbers = new ArrayList<>();
+
+    public LottoAutoNumberGenerator() {
+        for (int number = Number.LOWER_LIMIT; number <= Number.UPPER_LIMIT; number++) {
+            candidateNumbers.add(Number.from(number));
+        }
+    }
+
+    @Override
+    public synchronized List<Number> generateNumbers() {
+        Collections.shuffle(candidateNumbers);
+        Set<Number> lottoNumbers =
+            new TreeSet<>(candidateNumbers.subList(0, Lotto.POSSESSION_NUMBER));
+        return new ArrayList<>(lottoNumbers);
```

Comment:
```suggestion
        final List<Number> numbers = new ArrayList<>(candidateNumbers);
        Collections.shuffle(numbers);
        Set<Number> lottoNumbers =
            new TreeSet<>(numbers.subList(0, Lotto.POSSESSION_NUMBER));
        return new ArrayList<>(lottoNumbers);
```

해당 코드와 어떠한 차이가 있을지 고민해보아요

Reply:
각각의 작업을 할 때마다, 새로운 리스트를 만들어서 복사하여, 최대한 쓰레드 간의 간섭을 막는 것 같습니다!
이전의 static일 때는, 클래스 변수로서 공동인 메모리에 들어가서, 쓰레드 간섭의 위험이 있었는데,
이번에는 각각의 인스턴스 별로 인스턴스 변수를 가져서, 쓰레드간의 충돌이 없을줄 알았는데,
그래도 복사해서 사용을 해야하는건가요?

Reply:
아니면 현 상황에서는 한 인스턴스가 한 사용자(쓰레드)에게만 가서 상관이 없지만,
나중에 인스턴스가 공유되면, 일어날 불상사를 막는 코드인지 궁금합니다!